classes/Product.php

```diff
@@ -298,7 +297,7 @@ class ProductCore extends ObjectModel
      *
      * @var bool|null
      */
-    public $depends_on_stock;
+    public $depends_on_stock = false;
```

Comment:
All good for the other modification for me, but I think you can go one step further and remove this field along with `advanced_stock_management` since they have been deprecated since 1.7.8

We could also use the opportunity to remove these columns from the database

Reply:
Done :-)